File: packages/shared/src/types/message.ts
import type { UUID } from './common';

export interface Message {
  id: UUID;
  sender_id: UUID;
  sender_name: string;
  recipient_id: UUID;
  recipient_name: string;
  subject: string;
  body: string;
  thread_id: UUID | null;
  is_read: boolean;
  created_at: string;
}

export interface MessageThread {
  id: UUID;
  subject: string;
  participants: { id: UUID; name: string }[];
  last_message: Message;
  unread_count: number;
}

export interface MessageSend {
  recipient_id: UUID;
  subject: string;
  body: string;
  thread_id?: UUID;
}

File: apps/api/app/schemas/message.py
from pydantic import BaseModel, Field


class MessageSend(BaseModel):
    recipient_id: str
    subject: str = Field(min_length=1, max_length=200)
    body: str = Field(min_length=1)
    thread_id: str | None = None


class MessageOut(BaseModel):
    id: str
    sender_id: str
    sender_name: str | None = None
    recipient_id: str
    recipient_name: str | None = None
    subject: str
    body: str
    thread_id: str | None
    is_read: bool
    created_at: str

    model_config = {"from_attributes": True}


class ThreadOut(BaseModel):
    id: str
    subject: str
    participants: list[dict]
    last_message: MessageOut
    unread_count: int


class ThreadListOut(BaseModel):
    data: list[ThreadOut]
    total: int
